docs/rules/no-describe-variables.md

```diff
@@ -0,0 +1,71 @@
+# Disallow variables in describe blocks (no-describe-variables)
+
+Using variables in describe blocks in jasmine can cause memory leaks, e.g.:
+
+```js
+describe('Memory leak', function() {
+
+  var a;
+
+  beforeEach(function() {
+    a = new Array(10000000).join('a');
+  });
+
+  it('test', function() {
+    expect(a).toBeDefined();
+  });
+});
+```
+
+Due to internal design of the library such variables can not be collected by the 
+garbage collector. Instead the suggested pattern to use is:
+
+```js
+describe('Memory leak', function() {
+
+  beforeEach(function() {
+    this.a = new Array(10000000).join('a');
+  });
+
+  it('test', function() {
+    expect(this.a).toBeDefined();
+  });
+});
+```
+
+```js
+describe('Memory leak', function() {
+
+  var a;
+
+  beforeEach(function() {
+    a = new Array(10000000).join('a');
+  });
+  
+  afterEach(function () {
+    a = null; 
+  });
+
+  it('test', function() {
+    expect(a).toBeDefined();
+  });
+});
+```
+
+or
+
+## Rule details
+
+The following are considered warnings:
```

Comment:
"The following are considered warnings/The following patterns are not warnings" structure should be used for all exmaples